### PR TITLE
[Coil][Glide] Update docs for recent changes

### PR DESCRIPTION
**Pull request by chrisbanes**

### CODE REVIEW

File: docs/coil.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+# Coil for Jetpack Compose
+
+[![Maven Central](https://img.shields.io/maven-central/v/com.google.accompanist/accompanist-coil)](https://search.maven.org/search?q=g:com.google.accompanist)
+
+This library brings easy-to-use [Painter][painter] which can fetch and display images from external sources, such as network, using the [Coil][coil] image loading library.
```
Comment (nickbutcher) [inline docs/coil.md:5]: provides? offers?
```suggestion
This library provides easy-to-use [Painter][painter] which can fetch and display images from external sources, such as network, using the [Coil][coil] image loading library.
```

File: docs/coil.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+The primary API is via the [`rememberCoilPainter()`][rememberpainter] function. The simplest usage is like so:
+
+```kotlin 
+import androidx.compose.foundation.Image
+import com.google.accompanist.coil.rememberCoilPainter
+
+Image(
+    painter = rememberCoilPainter("https://picsum.photos/300/300"),
+    contentDescription = "My content description",
```
Comment (nickbutcher) [inline docs/coil.md:22]: What about using `stringResource` to promote best practice? Worth it?

File: docs/coil.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        "https://picsum.photos/300/300",
+        fadeIn = true
+    ),
+    contentDescription = "My content description",
+)
+```
+
+## Custom content
+
+Some times you may wish to display some alternative content whilst the image is loading, or an error has occurred. The painter returned from `rememberCoilPainter()` is an instance of [`LoadPainter`][loadpainter], which is stateful and allows you to display different content as required:
```
Comment (nickbutcher) [inline docs/coil.md:69]:
```suggestion
Some times you may wish to display alternative content whilst the image is loading, or an error has occurred. The painter returned from `rememberCoilPainter()` is an instance of [`LoadPainter`][loadpainter], which is stateful and allows you to display different content as required:
```

File: docs/coil.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+``` kotlin
+val painter = rememberCoilPainter("https://picsum.photos/300/300")
+
+Box {
+    Image(
+        painter = painter,
+        contentDescription = "My content description",
+    )
+
+    if (painter.loadState == ImageLoadState.Loading) {
```
Comment (nickbutcher) [inline docs/coil.md:81]: Use a `when` here to show the other states (with ellided bodys)?

File: docs/coil.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+```kotlin
+Image(
+    painter = rememberCoilPainter(
+        request = "https://picsum.photos/300/300",
+        previewPlaceholder = R.drawable.placeholder,
+    ),
+    contentDescription = "My content description",
+)
+```
+
```
Comment (nickbutcher) [inline docs/coil.md:103]: Worth calling out that preview placeholders don't need to be in the main source set?

File: docs/coil/migration-coilimage.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
+# Migration from CoilImage
+
+In Accompanist v0.8.0 the Coil library was refactored, moving away from the `CoilImage()` functions, to a new remembered painter provided by [`rememberCoilPainter()`][rememberpainter]. This page details how to migrate, as well as alternatives for function which no longer exists.
+
+## CoilImage is now deprecated
+The `CoilImage()` functions still exist (as of v0.8.0) but are now deprecated. They will be removed in a future release of Accompanist (likely v1.0).
+
+### ReplaceWith
+
+An automatic replacement migration has been provided in the deprecation. This allows you to migrate one or all `CoilImage()` calls over to the new API.  This works perfectly in most cases but you may find that the resulting code needs tidying up, YMMV.
```
Comment (nickbutcher) [inline docs/coil/migration-coilimage.md:10]:
```suggestion
An automatic replacement migration has been provided in the deprecation. This allows you to migrate one or all `CoilImage()` calls over to the new API.
```

File: docs/coil/migration-coilimage.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+## Removed features
+
+The new API is intentionally missing some functionality which was previously available in `CoilImage`. Migrating usages of those features requires manual migration:
+
+### onRequestCompleted()
+
+`CoilImage` previously allowed the passing of a lambda which was invoked whenever a request was completed. This has been removed to simplify the internal state. If you wish to convert state changes into events, you can use [`snapshotFlow()`][snapshotflow] to observe changes to `painter.loadState`, and then call your logic as necessary:
```
Comment (nickbutcher) [inline docs/coil/migration-coilimage.md:18]: Should this be in the main coil doc, not just in the migration guide? Imagine this page will go away at some point.

**Review by nickbutcher — APPROVED** [review]
A few small suggestions otherwise LGTM.

File: docs/coil/migration-coilimage.md (hunk 7 above)
Comment (chrisbanes) [reply docs/coil/migration-coilimage.md:18]: Done

File: docs/coil.md (hunk 5 above)
Comment (chrisbanes) [reply docs/coil.md:103]: Done

File: docs/coil.md (hunk 4 above)
Comment (chrisbanes) [reply docs/coil.md:81]: Done

File: docs/coil.md (hunk 2 above)
Comment (chrisbanes) [reply docs/coil.md:22]: Done

**Pull request merged**